akka-projection-slick/src/main/scala/akka/projection/slick/internal/SlickProjectionImpl.scala

```diff
@@ -48,14 +50,37 @@ private[projection] class SlickProjectionImpl[Offset, Envelope, P <: JdbcProfile
 
   private val killSwitch = KillSwitches.shared(projectionId.id)
   private val promiseToStop: Promise[Done] = Promise()
-
   private val started = new AtomicBoolean(false)
 
+  override def withSettings(projectionSettings: ProjectionSettings): Projection[Envelope] = {
+    new SlickProjectionImpl(projectionId, sourceProvider, databaseConfig, strategy, Option(projectionSettings), handler)
+  }
+
   override def run()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
     if (started.compareAndSet(false, true)) {
-      val done = mappedSource().runWith(Sink.ignore)
+      val done = mappedSource().run()
+      promiseToStop.completeWith(done)
+    }
+  }
+
+  override def runWithBackoff()(implicit systemProvider: ClassicActorSystemProvider): Unit = {
+
+    val projectionSettings = projectionSettingsOpt.getOrElse(ProjectionSettings(systemProvider))
+
+    if (started.compareAndSet(false, true)) {
+      val done =
+        RestartSource
+          .withBackoff(
```

Comment:
Shouldn't it be `RestartSource.onFailuresWithBackoff` so that it's only restarted when there is a failure. Also means that it will not be restarted by the killSwitch inside mappedSource, i.e. the outer killSwitch can be removed.

Reply:
Lagom use `withBackoff`, that's why I transposed it here. 

If we assume that a Projection only makes sense if its an infinite source, then a source that completes should be restarted.  

I may consume from a source that do complete and for each I don't have control over, then I can add a backoff of 2 hours and resume consuming even later on in the hope that it will offer me new elements. If it completes, I retry in 2 hours and so on.

Reply:
ok, but then we need to work on where to place the killSwitch. I think it's wrong to place it at several places. I think things are teared down more gracefully if the killswitch is in the beginning of the flow. Otherwise the Sink mat value might be completed before current element has been finished processing.

Reply:
We decided to use `onFailuresWithBackoff`. Sources that are finite, will need to be wrapped on an `RestartSource` by the implementers.